src/tui.rs

```diff
@@ -14,12 +14,131 @@ use ratatui::{
+            Component::AssetClass,
+            Component::Amount,
+            Component::Balance,
+        ]
+    }
+
+    /// Returns the string representation of the component
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Component::TabBar => "tab_bar",
+            Component::TotalValue => "total_value",
+            Component::AssetAllocation => "asset_allocation",
+            Component::DetailedAllocation => "detailed_allocation",
+            Component::Help => "help",
+            Component::Name => "name",
+            Component::AssetClass => "asset_class",
+            Component::Amount => "amount",
+            Component::Balance => "balance",
+        }
+    }
+
+    /// Returns a description of what the component does
+    pub fn description(&self) -> &'static str {
+        match self {
+            Component::TabBar => "Top navigation bar showing active tab",
+            Component::TotalValue => "Total portfolio value display",
+            Component::AssetAllocation => "Asset bar chart",
+            Component::DetailedAllocation => "Asset percentages",
+            Component::Help => "Keyboard shortcuts",
+            Component::Name => "Name column in the balances table",
+            Component::AssetClass => "Asset Class column in the balances table",
+            Component::Amount => "Amount column in the balances table",
+            Component::Balance => "Balance column in the balances table",
+        }
+    }
+}
+
+impl FromStr for Component {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.trim().to_lowercase().as_str() {
+            "tab_bar" => Ok(Component::TabBar),
+            "total_value" => Ok(Component::TotalValue),
+            "asset_allocation" => Ok(Component::AssetAllocation),
+            "detailed_allocation" => Ok(Component::DetailedAllocation),
+            "help" => Ok(Component::Help),
+            "name" => Ok(Component::Name),
+            "asset_class" => Ok(Component::AssetClass),
+            "amount" => Ok(Component::Amount),
+            "balance" => Ok(Component::Balance),
+            _ => Err(format!("Unknown component: '{s}'")),
+        }
+    }
+}
+
+impl std::fmt::Display for Component {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct DisabledComponents {
+    disabled: HashSet<Component>,
+}
+
+impl DisabledComponents {
+    pub fn new(disabled_list: Vec<String>) -> Self {
+        let mut disabled = HashSet::new();
+
+        for component_str in disabled_list {
+            match Component::from_str(&component_str) {
+                Ok(component) => {
+                    disabled.insert(component);
+                }
+                Err(err) => eprintln!("Warning: {err}"),
+            }
+        }
+
+        DisabledComponents { disabled }
+    }
```

Comment:
Consider returning a `Result<DisabledComponents, Vec<String>>` instead:

```rust
impl DisabledComponents {
    pub fn new(disabled_list: Vec<String>) -> Result<Self, Vec<String>> {
        let mut disabled = HashSet::new();
        let mut errors = Vec::new();

        for component_str in disabled_list {
            match Component::from_str(&component_str) {
                Ok(component) => { disabled.insert(component); }
                Err(err) => errors.push(err),
            }
        }

        if errors.is_empty() {
            Ok(DisabledComponents { disabled })
        } else {
            Err(errors)
        }
    }
}
```

This would make the validation explicit and testable while maintaining the same functionality.